docs/upgrade.html

```diff
@@ -36,9 +36,9 @@ <h5><a id="upgrade_370_notable" href="#upgrade_370_notable">Notable changes in 3
         </li>
     </ul>
 
-<h4><a id="upgrade_3_6_0" href="#upgrade_3_6_0">Upgrading to 3.6.0 from any version 0.8.x through 3.5.x</a></h4>
+<h4><a id="upgrade_3_6_1" href="#upgrade_3_6_1">Upgrading to 3.6.1 from any version 0.8.x through 3.5.x</a></h4>
```

Comment:
Note there is a slight divergence in philosophy here. @mimaison mentioned that it's best to have the latest version in this upgrade section, whereas with 3.5.2 and 3.5.1 we had a mini section mentioning the extra stuff from those versions and linking to the major 3.5.0 upgrade notes. Not sure what we wanna do going forward, but sticking to this since it's what we have as content right now

Reply:
FWIW I agree with @mimaison's suggestion. Patch versions are "patches". Their upgrade path and associated impact should be same as the associated minor version's upgrade. hence, it makes sense to only keep the upgrade documentation at a minor version level. 

I agree that this needs to be tackled separately from 3.7 release.

Reply:
Thanks for the cleanup!

We should revamp the upgrading section. This page is now huge as we keep adding to it for each release. I'm not sure keeping a section for all previous versions in the latest docs is really helpful. The upgrade steps already exist for each version in the docs for that version. Maybe we should still keep a section for each bugfix to avoid breaking links as we started linking to them in the blog page.

Reply:
I created https://issues.apache.org/jira/browse/KAFKA-16292 to refresh the upgrade.html file.

Reply:
thanks